src/System.Numerics.Tensors/src/System/Numerics/Tensors/ITensor.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Numerics.Tensors
+{
+    public interface ITensor<T>
+    {
+        ReadOnlySpan<int> Dimensions { get; }
+        bool IsReversedStride { get; }
+        long Length { get; }
+        int Rank { get; }
+        ReadOnlySpan<int> Strides { get; }
+
+        T this[params int[] indices] { get; set; }
```

Comment:
Should we support slicing such that we return a `Tensor` with a smaller rank?

Reply:
We had slicing before but it was removed when this code was moved to CoreFx.  Here's the issue that describes it: https://github.com/dotnet/corefxlab/issues/1819.

Reply:
@eerhardt, do you mean we need a `Range` type so that users can easily specify the slice to take from each dimension?

Reply:
Not "easily specify the slice", but just "specify the slice". If we didn't have some sort of `Range` type, how would we do it? At a minimum you need a list of `(int, int)` tuples. At that point - the `(int, int)` tuple **is** the "some sort of `Range` type".

But using the simple tuple design above is not ideal because: are the two ints `start` and `length` or `start` and `end? What if I want everything from index `5` to the end, but I don't know the end? Those are all things that the actual `Range` type is representing, and duplicating the behavior with an `(int, int)` tuple is non-ideal.